File: src/java/org/voegtle/weatherstation/server/request/IncomingUrlParameter.java
package org.voegtle.weatherstation.server.request;

import javax.servlet.http.HttpServletRequest;

public class IncomingUrlParameter extends AbstractUrlParameter {
  private static final String PARAM_DATA = "data";
  private static final String PARAM_LOCATION = "location";
  private static final String PARAM_SECRET = "secret";


  private final String data;
  private final String location;
  private final String secret;

  public IncomingUrlParameter(final HttpServletRequest request) {
    super(request);

    this.data = getUrlParameter(PARAM_DATA);
    this.location = getUrlParameter(PARAM_LOCATION);
    this.secret = getUrlParameter(PARAM_SECRET);
  }

  public String getData() {
    return data;
  }

  public String getLocation() {
    return location;
  }

  public String getSecret() {
    return secret;
  }
}

File: src/main/java/org/voegtle/weatherstation/server/request/IncomingUrlParameter.kt
package org.voegtle.weatherstation.server.request

import jakarta.servlet.http.HttpServletRequest
import org.voegtle.weatherstation.server.util.DateUtil

class IncomingUrlParameter(request: HttpServletRequest, dateUtil: DateUtil) : AbstractUrlParameter(request, dateUtil) {

  private val PARAM_DATA = "data"
  private val PARAM_LOCATION = "location"
  private val PARAM_SECRET = "secret"

  val data: String? = getUrlParameter(PARAM_DATA)
  val location: String? = getUrlParameter(PARAM_LOCATION)
  val secret: String? = getUrlParameter(PARAM_SECRET)
}
